File: Remove Duplicates from Sorted Array - Leetcode 26/Remove Duplicates from Sorted Array - Leetcode 26.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        n = len(nums)
        j = 1
        for i in range(1, n):
            if nums[i] != nums[i-1]:
                nums[j] = nums[i]
                j += 1
        
        return j
        
        # Time: O(n)
        # Space: O(1)

File: Remove Duplicates from Sorted Array - Leetcode 26/Remove Duplicates from Sorted Array - Leetcode 26.js
var removeDuplicates = function(nums) {
    if (nums.length === 0) return 0;
    let j = 1;
    for (let i = 1; i < nums.length; i++) {
        if (nums[i] !== nums[i - 1]) {
            nums[j] = nums[i];
            j++;
        }
    }
    return j;
};
